CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -279,34 +285,93 @@ void SpectrumLoader::HandleFile(TFile *f, Progress *prog) {
                      &XSSyst_cv_tmp[syst_it]);
   }
 
+  // If we are in the FD then these variables are nonsense.
+  if (sr.isFD) {
+    sr.muon_contained = -1;
+    sr.muon_tracker = -1;
+  }
+
   int Nentries = tr->GetEntries();
-  if (max_entries != 0 && max_entries < Nentries)
+  if (max_entries != 0 && max_entries < Nentries) {
     Nentries = max_entries;
+  }
+
+  TTree *potFriend = 0;
+  f->GetObject("OffAxisWeightFriend", potFriend);
+  if (potFriend) {
+    tr->AddFriend(potFriend);
+    SetBranchChecked(potFriend, "perPOT", &sr.perPOTWeight);
+    SetBranchChecked(potFriend, "perFile", &sr.perFileWeight);
+    SetBranchChecked(potFriend, "massCorr", &sr.NDMassCorrWeight);
+    SetBranchChecked(potFriend, "specRunWght", &sr.SpecialRunWeight);
+    SetBranchChecked(potFriend, "specRunId", &sr.SpecialHCRunId);
+ 
+    std::cout << "[INFO]: Found Off axis weight friend tree "
+                 "in input file, hooking up!"
+              << std::endl;
+  } else {
+    std::cout << "[WARNING]: Off axis weightings NOT being set." << std::endl;
+    sr.perPOTWeight = 1;
+    sr.perFileWeight = 1;
+    sr.NDMassCorrWeight = 1;
+    sr.SpecialRunWeight = 1;
+    sr.SpecialHCRunId = 293;
+  }
 
   for (int n = 0; n < Nentries; ++n) {
     tr->GetEntry(n);
 
+    if (!sr.isFD) sr.abspos_x = sr.det_x + sr.vtx_x;
+    else sr.abspos_x = 0;
+
     // Set GENIE_ScatteringMode and eRec_FromDep
     if (sr.isFD) {
-      sr.eRec_FromDep = sr.eDepP + sr.eDepN + sr.eDepPip +
-                        sr.eDepPim + sr.eDepPi0 +
-                        sr.eDepOther + sr.LepE;
+      sr.eRec_FromDep = sr.eDepP + sr.eDepN + sr.eDepPip + sr.eDepPim +
+                        sr.eDepPi0 + sr.eDepOther + sr.LepE;
```

Comment:
Loads of meaningless whitespace changes in this file which I suspect are going to cause me big merge conflicts :(

Reply:
Sorry :(